pkg/flags/flags.go

```diff
@@ -34,6 +38,13 @@ func setLogLevel(logLevel string) {
 		log.Fatalf("invalid log-level: %s", logLevel)
 	}
 	log.SetLevel(level)
+
+	klog.SetOutput(ioutil.Discard)
```

Comment:
No reason to do this. But you will want to set the log file to `/dev/null`.

Reply:
While working on this I found that any flags that were set in this method would not have any effect on the logs because we would be modifying flags after `flags.Parse()` was called. So setting the `log_file` to `/dev/null` would not work here.  Setting the output to `ioutil.DIscard`seemed to work.

Reply:
Looks like it is because you created the second flag set. This works:

```
func ConfigureAndParse() {
	// -stderrthreshold=FATAL forces klog to only log FATAL errors to stderr.
	// -logtostderr to false to not log to stderr by default.

	klog.InitFlags(nil)
	flag.Set("stderrthreshold", "FATAL")
	flag.Set("logtostderr", "false")
	flag.Set("log_file", "/dev/null")
	flag.Set("v", "0")
	logLevel := flag.String("log-level", log.InfoLevel.String(),
		"log level, must be one of: panic, fatal, error, warn, info, debug")
	printVersion := flag.Bool("version", false, "print version and exit")

	flag.Parse()

	setLogLevel(*logLevel)
	maybePrintVersionAndExit(*printVersion)
}

func setLogLevel(logLevel string) {
	level, err := log.ParseLevel(logLevel)
	if err != nil {
		log.Fatalf("invalid log-level: %s", logLevel)
	}
	log.SetLevel(level)

	// Anything lower than the INFO level according to log is sent to /dev/null
	if level == log.DebugLevel {
		flag.Set("stderrthreshold", "INFO")
		flag.Set("v", "10")
	}
}
```